kafka/integration_test.go

```diff
@@ -386,6 +440,22 @@ func consumerTest(t *testing.T, testname string, msgcnt int, cc consumerCtrl, co
 
 	}
 
+	if cc.withStats {
+		//wait for a while for stats to show up
+		for i := 0; i < 3; i++ {
+			if mt.statscnt != 0 {
```

Comment:
Is this thread-safe? mt.statscnt is updated through another go-routine, right?
I suggest changing this to a channel instead to signal completion.
E.g., let the stats receiving go-routine count the number of stats and send a "done" message on a channel when it is done, which is waited on here (with a reasonable timeout).